packages/math/src/pool/concentrated/one-for-zero.ts

```diff
@@ -0,0 +1,111 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+
+import { maxSqrtRatio } from "./const";
+import { SwapStrategy } from "./swap-strategy";
+import {
+  calcAmount0Delta,
+  calcAmount1Delta,
+  getAmountRemainingLessFee,
+  getFeeChargePerSwapStepOutGivenIn,
+  getNextSqrtPriceFromAmount1RoundDown,
+} from "./utils";
+
+export type OneForZero = {
+  isOutGivenIn: boolean;
+  sqrtPriceLimit: Dec;
+  swapFee: Dec;
+};
+
+export class OneForZeroStrategy implements SwapStrategy {
+  constructor(private readonly oneForZero: OneForZero) {}
+
+  computeSwapStep(
```

Comment:
Note: this function is going to be updated chain-side after refactor. We can work together next week on the updates